[docs] Add SECURITY.md (#15804)
mui-org/material-ui: SECURITY.md

# Security Policy

## Supported Versions

The versions of the project that are currently supported with security updates.

| Version | Supported          |
| ------- | ------------------ |
| 4.x | :white_check_mark: |
| 3.x | :white_check_mark: |
| < 3.0 | :x: |

## Reporting a Vulnerability

> You can report a vulnerability sending us an email at material.ui.org@gmail.com.

Reviewer: ```suggestion
You can report a vulnerability by sending us an email at material.ui.org@gmail.com.
```

> You can report a vulnerability sending us an email at material.ui.org@gmail.com.

Reviewer: ```suggestion
You can report a vulnerability by contacting us by email at material.ui.org@gmail.com.
```

Reviewer 2: Maybe `by contacting us via email` to reduce overuse of `by`

> You can report a vulnerability following the instructions in https://tidelift.com/docs/security.

Reviewer 3: So they're reported to tidelift and tidelift reports it to whom?

Author: Tidelift reports it to us, after filtering the noise. What alternative do we have? Do you prefer that we use our official email? `material.ui.org@gmail.com`

Reviewer 3: > Tidelift reports it to us 

But how do they do that if not by email? I can't imagine Tidelift filtering issues for us for free.

I'll have to rewatch the demo but I thought GitHub had a report mechanism built-in.

Author: I can't see anything about it. Let's start simple: material.ui.org@gmail.com.

## Reporting a Vulnerability

> You can report a vulnerability sending us an email at material.ui.org@gmail.com.

Reviewer 3: Is it possible that we can tie this to our docs domain i.e. `security@material-ui.com`? Can work on a better address later. This is good for merge.

Also: Please forward emails to that address to me as well.

Author: 💯 on that!
